File: laravel_api/app/Models/Ocupacao.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

/**
 * Representa a entidade Ocupacao (Vínculo Pessoa-Cargo).
 */
class Ocupacao extends Model
{
    use HasFactory;

    protected $table = 'Ocupacao';
    protected $primaryKey = 'id_ocupacao';
    public $incrementing = true;

    protected $fillable = [
        'id_pessoa',
        'id_cargo',
        'id_portaria',
        'data_inicio',
        'data_fim',
        'mandato',
        'observacoes',
    ];

    // Relações de Chave Estrangeira
    public function pessoa()
    {
        return $this->belongsTo(Pessoa::class, 'id_pessoa', 'id_pessoa');
    }

    public function cargo()
    {
        return $this->belongsTo(Cargo::class, 'id_cargo', 'id_cargo');
    }

    public function portaria()
    {
        return $this->belongsTo(Portaria::class, 'id_portaria', 'id_portaria');
    }
}

File: api/models/ocupacao.py
from sqlmodel import SQLModel, Field, UniqueConstraint
from typing import Optional
from datetime import date, datetime

class Ocupacao(SQLModel, table=True):
    id_ocupacao: Optional[int] = Field(default=None, primary_key=True)
    id_pessoa: int = Field(foreign_key="pessoa.id_pessoa")
    id_cargo: int = Field(foreign_key="cargo.id_cargo")
    data_inicio: Optional[date] = None
    data_fim: Optional[date] = None
    mandato: int
    observacoes: Optional[str] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    __table_args__ = (
        UniqueConstraint("id_pessoa", "id_cargo", "data_inicio", "data_fim", name="uix_ocupacao"),
    )
